### DISCUSSION TITLE
Request - "External video player for Movies only" option

### DISCUSSION THREAD

**Discussion by pavichokche** [jellyfin/jellyfin-androidtv#3303] [General]
As it stands, the built-in player options are great for format compatibility, but they're not very fast to load and you can't customize almost anything about them, making them a bit...clunky. External players are much better in the latter category, so it would be great to have an external player set to default for TV shows and use the integrated player(s) by default for movies. The slower loading up and worse UX doesn't matter so much for a movie since you put it on once and don't touch it for a long time, while the better format compatibility is obviously crucial.

**Comment by pavichokche** [comment]
MORE than ever this would be amazing.
Since v17 the built-in player just can't seem to play most of my movies, either due to surround sound audio issues, or HDR formats, or both, it's basically useless. However for TV shows it's perfectly fine AND it works great with the Intro/credits Skipper plugin. 
So PLEASE let us just pick external player for 'everything' or 'movies only' or 'disabled'! Please please it shouldn't be too tricky to implement

**Comment by ACR-Jeff** [comment]
I don't think this would happen, It's more a personal preference/choice, Not everyone would want a default player for movies and an external for TV shows, It's easier to just have either default or an external for all, 
This sounds like an encoding issue in your library, If they play fine on TV shows, Yet not in movies, May need to check the video formats. Also check your external players settings. May need to adjust something within the external players settings for encoding etc.

**Reply by pavichokche** [reply]
That's why all I asked for is an OPTION to have the external player applied to movies only, not to force anything on anyone. The existing option to use no external player or external player for everything can stay as-is.
My files play fine on the mobile android app, and they used to mostly play fine on the Android TV app before version 17. Now, as the Issues page shows, the Android TV app is plagued with HDR/surround sound issues. External apps like Vimu or Nova players handle my movies mostly fine as well.

**Reply by ACR-Jeff** [reply]
Maybe try mpv player, It seems to work flawless for me, I had issues with videos playing at one time as well. I didn't have great results with nova in Jellyfin, But MPV just works for me. https://github.com/mpv-android/mpv-android

**Reply by pavichokche** [reply]
I tried it, but the interface appears made for cellphones, and it seemed unable to handle any dolby vision video. There also seemed to be nearly no customization or features to speak of. I had just grabbed one of the .apks from their github repository, I didn't try fdroid.
And again, I HAVE an external video player that I like and am happy with, but I would prefer to use it for movies only. For TV shows the built-in player is fine (they have no surround sound/HDR) and it allows me to use the awesome Intro Skipper plugin. Hence my simple request...

**Reply by ACR-Jeff** [reply]
Ok, Well I am just trying to help in what ever way possible. I don't use Dolby vision because my setup doesn't support it yet, There is also "Just (Video) Player" which supports Dolby Vision per their readme https://github.com/moneytoo/Player?tab=readme-ov-file#supported-formats. I know you say you have your choice in video player that you like. Here is also another list of open source software if that's what your looking for. https://github.com/Generator/Awesome-Android-TV-FOSS-Apps?tab=readme-ov-file#media-player

**Reply by pavichokche** [reply]
I've tried Just Player in the past, it's pretty nice as well. So far I haven't found anything that integrates with Jellyfin better than Vimu Player, the one I'm using right now. I think it's the only one that is able to inform Jellyfin of the video progress point when I exit it, instead of just marking things as watched/unwatched. And it integrates with Google Home, giving a notification with controls on my phone while a video is playing. The only thing it doesn't do is take commands/give feedback to the Jellyfin server, since it IS external, so the server doesn't know when something is playing and it's not able to send Intro Skipper commands to it.

I just want the option to not use it on TV Shows. The Jellyfin AndroidTV app seems to be able to know what is a Movie and what isn't, so being able to toggle "Use External Player for Movies Only" as an advanced option (defaulted to Off obviously) shouldn't be that complicated or controversial. There's tons of settings in this app that are about customization and personal preference.

**Reply by ACR-Jeff** [reply]
Oh ok, Gotcha, Makes sense now that you explain it more, Well I hope they can implement this feature for you, Maybe create a feature request?

**Comment by pavichokche** [comment]
This is still SORELY needed, now more than ever. The internal player is AMAZING for **_TV shows_** - now with trick play images and official segment skipping support **_I REALLY_** want to use it. But for movies the internal player is just not good enough - it straight up crashes on a bunch of formats or needs transcoding/remuxing for this thing or that, it's completely unpredictable and unreliable.
Having the **_option_** to use external player **_for movies only_** would solve these issues entirely and allow us to have the best of both worlds!

**Comment by pavichokche** [comment]
PLEASE please please consider somehow, I don't care how, allowing people to use external player (if they want) for movies and internal player for TV.
The internal player is NO GOOD for most of my movies. There is no rhyme or reason, some files transcode and work, some transcode but crash, some transcode partially but stop playing after a bit, some don't transcode and crash. Only external players work consistently for movies.
BUT the internal player is AWESOME for TV shows with intro/credits skipping, and segment thumbnails. So PLEASE let us do that somehow. I'm always either missing out on a ton of awesome features for TV, or unable to watch movies (and have to keep switching to external player).

**Comment by cch85** [comment]
This pull request might have help you for this by doing long press and selecting your player at play time (as it was the case in the past if I am not mistaken): https://github.com/jellyfin/jellyfin-androidtv/pull/4487
But it was rejected as the owner of the PR did not reply to comment. I was not sure why this was reviewer wrote this comment: "adding a new getExternalPlayer function is against the concept of getPlaybackDestination function in PlaybackLauncher supposed to choose the destination"

**Reply by pavichokche** [reply]
Oh dang, yeah this would be an excellent feature!

**Comment by BenMcLean** [comment]
External players aren't a movies vs TV shows scoped option conceptually.

They also aren't conceptually a "one universal setting" option like the app currently has them.

External players are conceptually an individual video setting and that's where they belong in the UI.

**Reply by pavichokche** [reply]
Yes, that makes sense. I think there are enough requests out there to add a simple "Play in External Player" button to all video items, so I'll mark this exact one as closed.
